### PR TITLE
Decontextualize the decontextualizable

### PR DESCRIPTION
**Pull request by gahr**

### CODE REVIEW

File: mbox/mbox.c [hunk 1]
```diff
@@ -1800,8 +1800,8 @@ static enum MxStatus mbox_mbox_check_stats(struct Mailbox *m, uint8_t flags)
                                                       &m->stats_last_checked) > 0)
   {
     bool old_peek = m->peekonly;
-    struct Context *ctx = mx_mbox_open(m, MUTT_QUIET | MUTT_NOSORT | MUTT_PEEK);
-    mx_mbox_close(&ctx);
+    mx_mbox_open(m, MUTT_QUIET | MUTT_NOSORT | MUTT_PEEK);
+    mx_mbox_close(m);
```
Comment (flatcap) [inline mbox/mbox.c:1804]: (your change didn't introduce this problem, but...)

We may need to preserve/restore some flags for code like this.
- `MUTT_QUIET` affects `Mailbox->verbose`
- `MUTT_PEEK` affects `Mailbox->peekonly`

File: context.c [hunk 2]
```diff
@@ -430,3 +438,22 @@ struct Mailbox *ctx_mailbox(struct Context *ctx)
 {
   return ctx ? ctx->mailbox : NULL;
 }
+
+/**
+ * ctx_set_mailbox - Handle the post-mx_mbox_open steps
+ * @param m Mailbox
+ * TODO - kill this
+ */
+struct Context *ctx_set_mailbox(struct Mailbox *m)
```
Comment (flatcap) [inline context.c:447]: Is this needed any more?

Comment (gahr) [reply context.c:447]: No, I haven't polished the branch yet, don't waste time on the details :)

**Comment by flatcap** [comment]
I've done a bit of tidying, so that I can more easily rebase the Index/Pager changes on top.

- Squash fixes into the original commits
- clang-format
- rebase over `master`

I've also split out the `ctx_mailbox(Context)` changes you made to `index/index.c`
My future Index changes will eliminate the `Context` entirely.

**Comment by flatcap** [comment]
rebased over `master`

**Review by flatcap — APPROVED** [review]

**Pull request merged**